lib/hyrax/transactions/steps/apply_visibility.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+module Hyrax
+  module Transactions
+    module Steps
+      ##
+      # A `dry-transaction` step that interprets visibility/lease/embargo from
+      # passed arguments.
+      #
+      # @since 3.0.0
+      class ApplyVisibility
+        include Dry::Transaction::Operation
+
+        ##
+        # @param [Hyrax::WorkBehavior] work
+        # @param [String] visibility
+        # @param [String] release_date
+        # @param [String] during
+        # @param [String] after
+        #
+        # @return [Dry::Monads::Result] `Failure` if there is no
+        #   `PermissionTemplate` for the input; `Success(input)`, otherwise.
+        def call(work, visibility: nil, release_date: nil, during: nil, after: nil)
+          return Success(work) unless visibility.present?
+
+          intention = Hyrax::VisibilityIntention.new(visibility:   visibility,
+                                                     release_date: release_date,
+                                                     during:       during,
+                                                     after:        after)
+
+          Hyrax::VisibilityIntentionApplicator.new(intention: intention).apply_to(work)
```

Comment:
How about adding the convenience methods that are in Hyrax::PermissionTemplateApplicator to Hyrax::VisibilityIntentionApplicator so that these call chains match and this line becomes:
`Hyrax::VisibilityIntentionApplicator.apply(intention).to(model: work)`
The two applicator service classes are so similar that it seems like they should conform to the same interface so there is a clear pattern for future classes and to ease in reading them.

Reply:
I like this. and will add it.

Reply:
added. I got some more unit tests in for this class in the meanwhile. Thanks for the feedback.